### DISCUSSION TITLE
How do you organise your GitHub repositories?

### DISCUSSION THREAD

**Discussion by pftnhr** [datenstrom/community#893] [Ask a question] [answered]
I'm interested in where you put your zip archives on GitHub so that they are not in the same repo as your files and how you get there.

Do you use workflows or actions to organise files, folders, etc. on GitHub?

**Comment by schulle4u** [comment]
The recommended method is to use individual repositories for each extension. This is required by the [self-review checklist](https://github.com/annaesvensson/yellow-publish/blob/main/self-review-checklist.md) if you are planning to publish your code officially. If your extension should stay experimental, you can simply reference the repositories auto-generated zip download in your documentation and installation settings, e.g. `https://github.com/pftnhr/yellow-my-extension/archive/main.zip`

Not speaking as a maintainer, I'm a little rebellious at this point and prefer one multi-extensions repository for various reasons. 😈 A workflow to create zip files indeed would be nice, for now I'm doing this manually and store them in a downloads folder inside my repo.

**Reply by pftnhr** [reply]
Is the folder where your zip's are located a private one?

**Reply by schulle4u** [reply]
Not sure if I correctly understand the question, but for my multi-repository it is a [normal public folder](https://github.com/schulle4u/yellow-extensions-schulle4u/tree/main/downloads).

**Reply by pftnhr** [reply]
I meant the zip for hello-world but Mark just explained it.

**Comment by markseuffert** [comment, marked as answer]
Use one repository per extension. A ZIP file is automatically created by GitHub. Users can now download the ZIP file and copy it into their `system/extensions` folder. Please note that ZIP files must contain a file called `extension.ini`, otherwise Datenstrom Yellow doesn't know what to do with it. [Learn more about making and publishing extensions](https://github.com/annaesvensson/yellow-publish).

**Reply by pftnhr** [reply]
Ah! I was not aware that GitHub creates the archives. The 'extension.ini' is of course included.

I will add the link to the zip later and then make the feature public.

**Reply by markseuffert** [reply]
![meme-this-is-the-way](https://user-images.githubusercontent.com/2429165/136006429-2cf84dab-9707-4b0c-91f9-d9803973fecf.jpg)

**Reply by pftnhr** [reply]
I have one more (hopefully the last) question: GitHub gives me `https://github.com/pftnhr/yellow-extension/archive/refs/heads/main.zip` as a link. Is there any way I can get rid of the `/refs/heads`?

**Reply by markseuffert** [reply]
It looks correct to me. You can take any download link that works, see also [small example feature](https://github.com/datenstrom/yellow-newfeature).

**Reply by pftnhr** [reply]
Ok, it's done: https://github.com/datenstrom/yellow/discussions/894
